pkg/apis/query/v0alpha1/query.go

```diff
@@ -30,11 +30,11 @@ type QueryDataResponse struct {
 // GetResponseCode return the right status code for the response by checking the responses.
 func GetResponseCode(rsp *backend.QueryDataResponse) int {
 	if rsp == nil {
-		return http.StatusBadRequest
+		return http.StatusTeapot // rsp is nil, so we return a teapot
```

Comment:
While I love a teapot easter egg, will this mean we miss errors in our slo?

Reply:
Our SLO is 500s over time so this won't change our SLO performance one way or another.

The teapot use is to indicate that these messages had no downstream status code. I realized we were previously masking real 400s we might get from the client or further downstream.

The new logic in https://github.com/grafana/grafana/pull/105496/commits/5574f4f4dbc3653c3f6fee0e996c0269bd2ede51 first checks for the combination of `Error AND Status`. If we have an `Error` but no `Status` we use a teapot.

Reply:
I guess it will still get converted to a 500 by the responder.Error... but we'd be logging out teapots so we just have to know what that means haha. But I'm game